.github/workflows/run-dev-tests.yml

```
@@ -0,0 +1,108 @@
+name: "Check python code"
+
+on:
+  push:
+    paths:
+      - "**.py"
+      - .pylintrc
+      - mypy.ini
+      - ".github/workflows/run-dev-tests.yml"
+  pull_request:
+    types: opened
+    paths:
+      - "**.py"
+      - .pylintrc
+      - mypy.ini
+      - ".github/workflows/run-dev-tests.yml"
+
+jobs:
+  check-python-sources:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.x'
+      - run: python3 -m pip install pre-commit
+      - run: pre-commit run --all-files
+      - name: Build wheel
+        run: python3 -m pip wheel --no-deps -w dist .
+      - name: Upload wheel as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: cpp-linter-action_wheel
+          path: ${{ github.workspace }}/dist/*.whl
+
+  run-pytest:
+    needs: [check-python-sources]
+    strategy:
+      fail-fast: false
+      matrix:
+        py: ['3.7', '3.8', '3.9', '3.10']
+        os: ['windows-latest', ubuntu-latest]
+        version: ['13', '12', '11', '10']
```

Do we need to include old versions like 9, 8, 7, etc for testing?

---

```suggestion
        version: ['13', '12', '11', '10', '9', '8', '7']
```

I only didn't add those because we never supported those versions with the docker (& it would mean longer workflow runs with them added).

---

the versions are now added

---

It looks like the YML output from clang-tidy had breaking changes in v9. Meaning our current code needs to support the older versions of clang-tidy's YML output.

This isn't actually used in deployment, but I set up unit tests to check if we can parse the YML output. This will be beneficial if we ever support posting suggestions from clang-tidy in the event's diff.

---

Versions 7 & 8 are now passing the unit tests. Historically, this is a new feature because we have not supported v7 & v8 in the docker.

I'd update the README about this, but the README is rather specific to using the docker. This new unit test workflow doesn't use the docker though.